tests/test_span_readable.py

```diff
@@ -0,0 +1,65 @@
+import time
+from instana.span.readable_span import Event, ReadableSpan
+from instana.span_context import SpanContext
+from opentelemetry.trace.status import Status, StatusCode
+
+from instana.w3c_trace_context.traceparent import Traceparent
+
+
+def test_readablespan(
+    span_context: SpanContext,
+    trace_id: int,
+    span_id: int,
+) -> None:
+    span_name = "test-span"
+    timestamp = time.time_ns()
+    span = ReadableSpan(span_name, span_context)
+
+    assert span is not None
+    assert isinstance(span, ReadableSpan)
+    assert span.name == span_name
+
+    context = span.context
+    assert isinstance(context, SpanContext)
+    assert context.trace_id == trace_id
+    assert context.span_id == span_id
+
+    assert span.start_time
+    assert isinstance(span.start_time, int)
+    assert span.start_time > timestamp
+    assert not span.end_time
+    assert not span.attributes
+    assert not span.events
+    assert not span.parent_id
+    assert span.duration == 0
+    assert span.status._status_code == StatusCode.UNSET
+
+    assert not span.stack
+    assert span.synthetic is False
+
+
+def test_readablespan_with_params(
+        span_context: SpanContext,
+        trace_id: int,
+        span_id: int,
+) -> None:
+    span_name = "test-span"
+    tp = Traceparent()
+    traceparent_id = "00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01"
+    _, _, parent_id, _ = tp.get_traceparent_fields(traceparent_id)
+    start_time = time.time_ns()
+    end_time = time.time_ns()
```

Comment:
You could also check if the duration is the difference between end time and start time.

Reply:
duration variable never changes inside Readable Span, is it normal?

Reply:
I figured out that I was working on different branch version. Now I pulled the last version of otel_migration, it's okay now